programs/nft-voter/src/state/registrar.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{
+    error::NftVoterError,
+    id,
+    state::CollectionConfig,
+    tools::{spl_token::get_spl_token_amount, token_metadata::get_token_metadata_for_mint},
+};
+use anchor_lang::prelude::*;
+use spl_governance::tools::spl_token::{get_spl_token_mint, get_spl_token_owner};
+
+/// Registrar which stores NFT voting configuration for the given Realm
+#[account]
+#[derive(Debug, PartialEq)]
+pub struct Registrar {
+    /// spl-governance program the Realm belongs to
+    pub governance_program_id: Pubkey,
+
+    /// Realm of the Registrar
+    pub realm: Pubkey,
+
+    /// Governing token mint the Registrar is for
+    /// It can either be the Community or the Council mint of the Realm
+    /// When the plugin is used the mint is only used as identity of the governing power (voting population)
+    /// and the actual token of the mint is not used
+    pub governing_token_mint: Pubkey,
+
+    /// MPL Collection used for voting
+    pub collection_configs: Vec<CollectionConfig>,
+
+    /// Reserved for future upgrades
+    pub reserved: [u8; 128],
+}
+
+/// Returns Registrar PDA seeds
+pub fn get_registrar_seeds<'a>(
+    realm: &'a Pubkey,
+    governing_token_mint: &'a Pubkey,
+) -> [&'a [u8]; 3] {
+    [b"registrar", realm.as_ref(), governing_token_mint.as_ref()]
+}
+
+/// Returns Registrar PDA address
+pub fn get_registrar_address(realm: &Pubkey, governing_token_mint: &Pubkey) -> Pubkey {
+    Pubkey::find_program_address(&get_registrar_seeds(realm, governing_token_mint), &id()).0
+}
+
+impl Registrar {
+    pub fn get_collection_config(&self, collection: Pubkey) -> Result<&CollectionConfig> {
+        return self
+            .collection_configs
+            .iter()
+            .find(|cc| cc.collection == collection)
+            .ok_or_else(|| NftVoterError::CollectionNotFound.into());
+    }
+}
+
+/// Resolves vote weight and voting mint for the given NFT
+pub fn resolve_nft_vote_weight_and_mint(
```

Comment:
This is probably OK, but the function relies on the fact that metaplex metadata only exists in one place for a mint and is not possible to spoof.  I don't know enough about metaplex to be totally sure of that

Reply:
I'm going to double check with the `Metaplex` crew, but the Metadata is PDA and hence must be unique and in addition we check the `collection.verified flag`